synfig-studio/src/gui/dialogs/canvasresize.cpp

```diff
@@ -0,0 +1,297 @@
+#else
+#ifdef HAVE_CONFIG_H
+#	include <config.h>
+#endif
+
+#include "canvasresize.h"
+
+#include <gui/app.h>
+#include <gui/dialogs/canvasproperties.h>
+#include <gui/localization.h>
+#include <gui/resourcehelper.h>
+#include <gui/widgets/widget_link.h>
+
+#endif
+
+using namespace etl;
+using namespace Gtk;
+using namespace sigc;
+using namespace std;
+using namespace studio;
+using namespace synfig;
+using namespace synfigapp;
+
+CanvasResize::CanvasResize(
+		Window &parent, handle<CanvasInterface> &ci,
+		CanvasProperties &cp, bool ratio_wh_toggle_status):
+	Dialog            ("", parent),
+	canvas_interface  (ci),
+	rend_desc         (ci->get_canvas()->rend_desc()),
+	canvas_properties (cp),
+	builder           (Builder::create_from_file(ResourceHelper::get_ui_path("canvas_resize.glade"))),
+	grid_content      (nullptr),
+	width             (nullptr),
+	height            (nullptr),
+	toggle_ratio_wh   (nullptr),
+	IS_TOGGLED_DEFAULT(ratio_wh_toggle_status),
+	is_toggled_default(true),
+	update_lock       (0)
+{
+	set_up_builder_widgets();
+	set_up_toggle_tooltips();
+
+	// Add builder grid content to this dialog window content
+	get_content_area()->add(*grid_content);
+
+	set_up_action_widgets();
+
+	set_resizable(false);
+
+	// Decide which dialog this is: "Resize with Content" or "Set Size"
+	set_up_dialog_label();
+	IS_TOGGLED_DEFAULT ? set_image_flags() : set_canvas_flags();
+
+	refresh_title();
+	refresh_content();
+
+	// Finally, connect all our signals
+	set_up_signals();
+}
+
+void CanvasResize::set_up_builder_widgets()
+{
+	builder->get_widget("grid_content", grid_content);
+	builder->get_widget("width"       , width);
+	builder->get_widget("height"      , height);
+	builder->get_widget("label"       , label);
+	builder->get_widget_derived("toggle_ratio_wh", toggle_ratio_wh);
+}
+
+void CanvasResize::set_up_toggle_tooltips()
+{
+	toggle_ratio_wh->property_tooltip_active()   = _("Unlink width and height");
+	toggle_ratio_wh->property_tooltip_inactive() = _("Link width and height");
+
+	// Force refresh new tooltip messages
+	toggle_ratio_wh->toggled();
+}
+
+void CanvasResize::set_up_action_widgets()
+{
+	add_button("_Advanced...", 1)->set_use_underline();
+	add_button("_Cancel"     , 2)->set_use_underline();
+	add_button("_OK"         , 3)->set_use_underline();
+}
+
+void CanvasResize::set_up_dialog_label()
+{
+	string dialog;
+
+	IS_TOGGLED_DEFAULT ? dialog = _("Image ") : dialog = _("Canvas ");
+
+	label->set_text(dialog + label->get_text());
+}
+
+void CanvasResize::set_up_signals()
+{
+	// Keep refreshing dialog when synfigapp::CanvasInterface object is updated
+	canvas_interface->signal_id_changed().connect(
+				mem_fun(*this, &CanvasResize::refresh_title));
+	canvas_interface->signal_rend_desc_changed().connect(
+				mem_fun(*this, &CanvasResize::on_rend_desc_changed));
+
+	// Widgets
+	width->signal_changed().connect(sigc::bind<SpinButton *>(
+				mem_fun(*this, &CanvasResize::on_spinbutton_updated), width));
```

Comment:
Continuing the above `using directive` review...

We have a collision here since both `std` and `sigc` have adaptor `bind` the programmer has to manually specify which `bind` he wants to use.

If only needed specific directive have been pulled (initial review above) this wouldn't be a problem.

Reply:
In other hand, no problem at all for explictly stating the namespace

Reply:
I checked out other synfig files and they seem to pull in only synfig project namespaces like `studio` and `synfig`.
I went the same way.

Otherwise, a readable line like this
```c++
...
	if (is_toggled)
		rend_desc.set_flags(rend_desc.get_flags() |  RendDesc::LINK_IM_ASPECT);
...
```
gets a bit lengthy and repetitive since there are multiple flags setting code going on in this dialog
```c++
...
	if (is_toggled)
		rend_desc.set_flags(rend_desc.get_flags() |  synfig::RendDesc::LINK_IM_ASPECT);
...
```

For other namespaces that are not part of synfig project/API, explicit qualifier naming are used.